src/Serialization/Rings.jl

```diff
@@ -1,3 +1,19 @@
+################################################################################
+# Utility functions for ring parent tree
+
+# builds parent tree
+function get_parents(parent_ring::Ring)
+    base = base_ring(parent_ring)
+    if has_elem_basic_encoding(base)
+        return Any[parent_ring]
+    end
```

Comment:
I don't quite understand what's going on here. Why isn't it
```suggestion
    if has_elem_basic_encoding(parent_ring)
        return Any[]
    end
    base = base_ring(parent_ring)
```

Reply:
... and then it would be basically identical to this one from above (so that method could be removed)
```julia

 function get_parents(parent_ring::T) where T <: Union{FracField,
                                                       AbstractAlgebra.Generic.RationalFunctionField,
                                                       AbstractAlgebra.Generic.LaurentSeriesField}
     if has_elem_basic_encoding(parent_ring)
         return Any[]
     end

     base = base_ring(parent_ring)
     parents = get_parents(base)
     push!(parents, parent_ring)
     return parents
 end
```